src/Reports/SalesByItems/Services/TotalsService.php

```diff
@@ -0,0 +1,198 @@
+    {
+        return $this->calculateTotals($items);
+    }
+
+    /**
+     * Calculate totals for a product
+     *
+     * @param SalesItem[] $items Items for the product
+     * @return array ['quantity' => int, 'total' => float]
+     */
+    public function calculateProductTotals(array $items): array
+    {
+        return $this->calculateTotals($items);
+    }
+
+    /**
+     * Calculate totals by category
+     *
+     * Returns array like:
+     * [
+     *   'Office Visit' => ['quantity' => 5, 'total' => 750.00],
+     *   'Products' => ['quantity' => 20, 'total' => 500.00],
+     * ]
+     *
+     * @param SalesItem[] $items Items to aggregate
+     * @return array Totals by category
+     */
+    public function calculateTotalsByCategory(array $items): array
+    {
+        $byCategory = [];
+
+        foreach ($items as $item) {
+            $category = $item->getCategory() ?: 'None';
+
+            if (!isset($byCategory[$category])) {
+                $byCategory[$category] = [];
+            }
+
+            $byCategory[$category][] = $item;
+        }
+
+        $totals = [];
+        foreach ($byCategory as $category => $categoryItems) {
+            $totals[$category] = $this->calculateTotals($categoryItems);
+        }
+
+        return $totals;
+    }
+
+    /**
+     * Calculate totals by category and product
+     *
+     * Returns array like:
+     * [
+     *   'Office Visit' => [
+     *       '99213' => ['quantity' => 2, 'total' => 300.00],
+     *       '99214' => ['quantity' => 3, 'total' => 450.00],
+     *   ],
+     *   'Products' => [
+     *       'Bandage' => ['quantity' => 20, 'total' => 500.00],
+     *   ],
+     * ]
+     *
+     * @param SalesItem[] $items Items to aggregate
+     * @return array Totals by category and product
+     */
+    public function calculateTotalsByCategoryAndProduct(array $items): array
+    {
+        $byGroup = [];
+
+        foreach ($items as $item) {
+            $category = $item->getCategory() ?: 'None';
+            $product = $item->getDescription() ?: 'Unknown';
+
+            if (!isset($byGroup[$category])) {
+                $byGroup[$category] = [];
+            }
+
+            if (!isset($byGroup[$category][$product])) {
+                $byGroup[$category][$product] = [];
+            }
+
+            $byGroup[$category][$product][] = $item;
+        }
+
+        $totals = [];
+        foreach ($byGroup as $category => $products) {
+            $totals[$category] = [];
+            foreach ($products as $product => $items) {
+                $totals[$category][$product] = $this->calculateTotals($items);
+            }
+        }
+
+        return $totals;
+    }
+
+    /**
+     * Format currency for display
+     *
+     * @param float $amount The amount to format
+     * @param string $format Format string (default: US format)
+     * @return string Formatted currency
+     */
+    public function formatCurrency(float $amount, string $format = '$%.2f'): string
```

Comment:
Currency format should be based on localization settings for the system, and should be handled in a utility service across all of openemr.